firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift

```diff
@@ -116,6 +113,30 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
     override func viewDidDisappear(_ animated: Bool) {
         super.viewDidDisappear(animated)
         setEditing(false, animated: false)
+        unsubscribeFromRedux()
+    }
+
+    // MARK: - Redux
+
+    func subscribeToRedux() {
+        store.dispatch(SearchSettingsAction.searchSettingsDidLoad)
+        store.subscribe(self, transform: {
+            $0.select(SearchSettingsState.init)
+        })
+    }
+
+    func unsubscribeFromRedux() {
+        store.unsubscribe(self)
+    }
+
+    func newState(state: SearchSettingsState) {
+        searchSettingsState = state
+        self.navigationItem.title = .Settings.Search.Title
+        
+        // To allow re-ordering the list of search engines at all times.
+        self.tableView.isEditing = true
+        // So that we push the default search engine controller on selection.
+        self.tableView.allowsSelectionDuringEditing = true
```

Comment:
Unless we're updating these here, we shouldn't be adding them ahead of time. Only when the state is actually doing something with them.